packages/dds/tree/src/test/edit-manager/editManager.spec.ts

```diff
@@ -200,536 +201,369 @@ function asDelta(intentions: number[]): Delta.Root {
+            { seq: 2, type: "Pull", ref: 0, from: peer2, expectedDelta: [-3, 2, 3] },
+            { seq: 6, type: "Push" },
+            { seq: 8, type: "Push" },
+            { seq: 3, type: "Ack" },
+            { seq: 4, type: "Pull", ref: 1, from: peer1, expectedDelta: [-8, -6, 4, 6, 8] },
+            { seq: 5, type: "Pull", ref: 2, from: peer1, expectedDelta: [-8, -6, 5, 6, 8] },
+            { seq: 6, type: "Ack" },
+            { seq: 7, type: "Pull", ref: 0, from: peer2, expectedDelta: [-8, 7, 8] },
+            { seq: 8, type: "Ack" },
+            { seq: 9, type: "Pull", ref: 0, from: peer2, expectedDelta: [9] },
+        ]);
     });
 
     /**
      * This test case effectively tests most of the scenarios covered by the other test cases.
      * Despite that, it's good to keep the other tests cases for the following reasons:
-     * - They give a clearer account of what the API usage is like.
-     * - They are easier to debug.
+     * - They are easier to read and debug.
      * - They help diagnose issues with the more complicated exhaustive test (e.g., if one of the above tests fails,
      *   but this one doesn't, then there might be something wrong with this test).
      */
-    it("Can handle all possible interleaving of steps", () => {
+    it("Combinatorial test", () => {
         const meta = {
-            clientData: makeArray(NUM_CLIENTS, () => ({ pulled: 0, numLocal: 0 })),
+            peerRefs: makeArray(NUM_PEERS, () => 0),
             seq: 0,
+            inFlight: 0,
         };
         for (const scenario of buildScenario([], meta)) {
-            // Uncomment the lines below to see which scenario fails first.
-            // const name = scenario.map((step) => `${step.type}${step.client}`).join("-");
-            // console.debug(name);
-            runScenario(scenario);
+            // Uncomment the code below to log the titles of generated scenarios.
+            // This is helpful for creating a unit test out of a generated scenario that fails.
+            // const title = scenario.map((s) => {
+            //     if (s.type === "Pull") {
+            //         return `Pull(${s.seq}) from:${s.from} ref:${s.ref}`;
+            //     } else if (s.type === "Ack") {
+            //         return `Ack(${s.seq})`;
+            //     }
+            //     return s.type;
+            // }).join("|");
+            // console.debug(title);
+            runUnitTestScenario(undefined, scenario);
         }
     });
 });
 
-type ScenarioStep =
-    // Represents a client making a local change
-    | { type: "Mint"; client: number; }
-    // Represents a change from a client being sequenced by the service
-    | { type: "Sequence"; client: number; }
-    // Represents a client receiving a sequenced change
-    | { type: "Receive"; client: number; }
-;
-
 /**
  * State needed by the scenario builder.
  */
 interface ScenarioBuilderState {
```

Comment:
A brief comment on the meaning of each field would make `buildScenario` easier to read.